### PR TITLE
SONARIAC-1452 Set the correct profile name

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

File: iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/SpringConfigProfileNameUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.springconfig.tree.api.Tuple;
+import org.sonar.iac.springconfig.tree.impl.ScalarImpl;
+import org.sonar.iac.springconfig.tree.impl.SyntaxTokenImpl;
+import org.sonar.iac.springconfig.tree.impl.TupleImpl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SpringConfigProfileNameUtilTest {
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/SpringConfigProfileNameUtilTest.java:31]: Basically to increase coverage.

File: iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitorTest.java [hunk 2]
```diff
@@ -247,8 +247,11 @@ static Stream<Arguments> shouldReadProfileName() {
     return Stream.of(
       arguments("spring.profiles.active=profile1", "profile1"),
       arguments("spring.config.activate.on-profile=dev & qa", "dev & qa"),
-      arguments("spring.profiles.active=profile1\nspring.config.activate.on-profile=profile2", "profile1 profile2"),
-      arguments("#comment", ""));
+      arguments("spring.profiles.active=profile1\nspring.config.activate.on-profile=profile2", "profile2"),
+      arguments("spring.profiles.active=profile1\nspring.profiles.active=profile2", "profile2"),
+      arguments("spring.profiles.default=profile1\nspring.profiles.active=profile2", "profile2"),
+      arguments("spring.profiles.default=profile1", "profile1"),
+      arguments("#comment", "default"));
```
Comment (mstachniuk) [inline iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitorTest.java:254]: Please add the following cases:
`"spring.config.activate.on-profile=profile1\nspring.profiles.active=profile2\n"`
`"spring.profiles.active=profile1\nspring.config.activate.on-profile=profile2\nspring.profiles.default=profile3"`

and maybe more.

Consider adding a space before/after `\n` for better readability.

**Review by mstachniuk — APPROVED** [review]
LGTM and please see my comment.

General question: How is it with `spring.profiles` property? Any influence on profile name?

File: iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitorTest.java (hunk 2 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitorTest.java:254]: My spring crashes when I try to use `spring.profiles`.

I added the test cases.
I also added a whitespace after the profile name.
This led me to trim the profile name after reading it (confirmed in my spring application).

Adding a space before a line resulted in a parsingError, I created a ticket for it (https://sonarsource.atlassian.net/browse/SONARIAC-1457)

**Pull request merged**